front-end-tools/src/reading_from_blockchain.ts

```diff
@@ -229,15 +228,14 @@ export async function read(
     const fullEntryPointName = `${contractName.value}.${entryPoint.value}`;
 
     if (!res || res.tag === 'failure') {
-        const rejectReason = JSON.stringify(
-            ((res as InvokeContractFailedResult)?.reason as RejectedReceive)?.rejectReason
-        );
+        const rejectReason = decodeRejectReason(res, contractName, entryPoint, moduleSchema);
 
         throw new Error(
             `RPC call 'invokeContract' on method '${fullEntryPointName}' of contract '${contractIndex}' failed.
             ${rejectReason !== undefined ? `Reject reason: ${rejectReason}` : ''}`
```

Comment:
We need to warn here that the prettyfied reject reason is not guaranteed to be correct.

There is no guarantee people don't override the error codes in their contract.